code-challenges.test.js

```diff
@@ -14,38 +15,259 @@
+    { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+    { name: "arthur dent", occupation: "a radio employee" }
+  ]
+
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized.", () => {
+
+      expect(employment(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
+
 
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+// const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+// // Expected output: [ 2, 0, -1, 0 ]
+// const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+// // Expected output: [ 2, 1, -1 ]
+
+
+//Pseudo Code:
+/*
+-Input an object array, the output is an array with remainder for numbers divided by 3.
+-Create a filter to remove non-numbers by performing a typeOf "number"
+-Perform a iteration for the filtered numbers, value % 3 
+-Output an array of the remainders for the values.
+-In addition, need to do a red-green-refactor methodolgy, people.
+*/
+
+// First Step: Perform Red
+
+// describe("remainders", () => {
+
+//   it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3", () => {
+
+//       expect(remainders(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+//   })
+// })
+
+// Received an error =>  ReferenceError: remainders is not defined
+
+//Second Step: Make the test pass for hodgepodge1, hodgepodge2
+
+// describe("remainders", () => {
+
+//   it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3", () => {
+
+//       expect(remainders(hodgepodge1)).toEqual([ 2, 0, -1, 0 ]);
+//       expect(remainders(hodgepodge2)).toEqual([ 2, 1, -1 ]);
+//   })
+// })
+
 
 
 // b) Create the function that makes the test pass.
+// -Create a filter to remove non-numbers by performing a typeOf "number"
+// -Perform a iteration for the filtered numbers, value % 3 
+// -Output an array of the remainders for the values.
+
+// const remainders = (arrayOfThings) => {
+//   arrayfilter = arrayOfThings.filter(value => {
+//     return typeof value === "number"
+//   }).map(value => value % 3)
+  
+  
+//   return arrayfilter
+// }
+
+//Third Step: Refactor
+/*
+Create implicit arrows to show .filter and .map in one line.
+*/
+
+const remainders = (arrayOfThings) => {
+  return arrayOfThings.filter(value => typeof value === "number").map(value => value % 3)
+}
+
+
+describe("remainders", () => {
+  const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
```

Comment:
It is good practice to put variables in the scope you are working in. In this case, the hodgepodge variables are used inside the it block. Just something to keep an eye out for when writing tests. Otherwise looks great!